File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.*;
import java.util.ArrayList;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        ArrayList<String> result = new ArrayList<>();
        Stack<Integer> stack = new Stack<>();
        int stackNum = 0;
        Boolean resultSwitch = true;
        loop: for (int i = 0; i < N; i++) {
            int inputNum = Integer.parseInt(br.readLine());
            if(stackNum < inputNum) {
                while (stackNum != inputNum) {
                    stackNum+=1;
                    stack.push(stackNum);
                    result.add("+");
                }
            }
            if (stackNum == inputNum) {
                stack.pop();
                result.add("-");
            } else if(stackNum > inputNum) {
                while (!stack.isEmpty()) {
                    int top = stack.peek();
                    stack.pop();
                    result.add("-");
                    if (top == inputNum) {
                        continue loop;
                    }
                }
                if (stack.isEmpty()) {
                    resultSwitch=false;
                }
            }
        }
        if (resultSwitch) {
            for (int i = 0; i < result.size(); i++) {
                bw.write(result.get(i)+"\n");
            }
            bw.flush();
        } else {
            System.out.println("NO");
        }
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
from sys import stdin as s

# input.txt file
#s = open('input.txt')

n = int(s.readline().rstrip())

item = 1
stack = []
result = []

for i in range(0, n):
    input = int(s.readline().rstrip())

    while item <= input:
        stack.append(item)
        item += 1
        result.append("+")

    if stack[-1] == input:
        stack.pop()
        result.append("-")
    else:
        print("NO")
        break
else:
    for i in result:
        print(i)